src/NHSD.BuyingCatalogue.Ordering.Persistence/Repositories/OrderRepository.cs

```diff
@@ -56,31 +55,31 @@ public async Task<string> CreateOrderAsync(Order order)
                 throw new ArgumentNullException(nameof(order));
             }
 
-            using (var dbContextTransaction = _context.Database.BeginTransaction())
+            using (var dbContextTransaction = _context.Database.BeginTransactionAsync())
             {
                 if (order.OrderId == null)
                 {
-                    order.OrderId = await GetIncremenedOrderId();
+                    order.OrderId = await GetIncrementedOrderId();
                 }
 
                 order.OrderStatus = await _context.OrderStatus.FindAsync(order.OrderStatus.OrderStatusId) ?? order.OrderStatus;
 
                 _context.Order.Add(order);
                 await _context.SaveChangesAsync();
-                await dbContextTransaction.CommitAsync();
+                await (await dbContextTransaction).CommitAsync();

```

Comment:
I think you'll be able to remove this `await` if you add one for `BeginTransactionAsync`.

Reply:
👍